KenticoCloud.Delivery/InlineContentItems/InlineContentItemsProcessor.cs

```diff
@@ -1,17 +1,30 @@
-﻿using System;
+﻿using AngleSharp.Dom;
+using AngleSharp.Parser.Html;
+using KenticoCloud.Delivery.Extensions;
+using System;
 using System.Collections.Generic;
 using System.Linq;
-using AngleSharp.Dom;
-using AngleSharp.Parser.Html;
 
 namespace KenticoCloud.Delivery.InlineContentItems
 {
+    /// <summary>
+    /// Resolves strongly typed model into string.
+    /// </summary>
+    /// <param name="o">Strongly typed model.</param>
+    /// <returns>String representation of the model (can be HTML, for instance).</returns>
+    internal delegate string ResolveInlineContent(object o);
+
     /// <summary>
     /// Processor responsible for parsing HTML input and resolving inline content items referenced in them using registered resolvers
     /// </summary>
     internal class InlineContentItemsProcessor : IInlineContentItemsProcessor
     {
-        private readonly Dictionary<Type, Func<object, string>> _typeResolver;
+        private  Dictionary<Type, ResolveInlineContent> _inlineContentItemsResolvers;
+        private Dictionary<Type, ResolveInlineContent> InlineContentItemsResolvers
+        {
+            get => _inlineContentItemsResolvers ?? (_inlineContentItemsResolvers = new Dictionary<Type, ResolveInlineContent>());
```

Comment:
That is a bit nasty. Since we use DI and we have the control over the default instance, we should probably not be over-protective towards dependencies…

As for tests, I believe, a factory that would ensure create instantiation in tests would be a bit better fitting 👼